pkg/registry/apis/provisioning/register.go

```diff
@@ -174,13 +163,25 @@ func (b *ProvisioningAPIBuilder) UpdateAPIGroupInfo(apiGroupInfo *genericapiserv
 		client: b.client.identities,
 		logger: b.logger.With("connector", "webhook"),
 	}
+	storage[provisioning.RepositoryResourceInfo.StoragePath("list")] = &listConnector{
```

Comment:
when you originally created the `/files` endpoint I thought that would implement this in a way that you return the list of files if the path finishes with `/` which indicates that it's a directory. Similar to what file servers do. 

Example omitting k8s: 

- GET /files/dashboards/
-> returns the list of files in the folder. 
```json
[
 "one.json",
 "second.json",
 "third.json" 
]
```

- GET /files/dashboards/one.json
return the dashboard
```json
{
  "title": "the first dashboard"
}
```

This would feel pretty natural to navigate the tree in the URL.